### PR TITLE
Some updates to Text 2 image pipeline

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
**TODO:**
- [ ] Python API and sample
- [ ] Update doc strings
- [x] Update main README.md (PR https://github.com/openvinotoolkit/openvino.genai/pull/930)
- [ ] Add sample with custom device mapping
- [ ] Experiment with reshape + compile as part of Ctor
- [x] Add LoRA (PR https://github.com/openvinotoolkit/openvino.genai/pull/911)
- [X] Use std::optional for prompt2, prompt3 and maybe negative prompts as well
- [X] Update https://github.com/openvinotoolkit/openvino.genai/blob/master/src/docs/SUPPORTED_MODELS.md with text 2 image generation models

### CODE REVIEW

File: samples/cpp/text2image/main.cpp [hunk 1]
```diff
@@ -44,10 +15,10 @@ int32_t main(int32_t argc, char* argv[]) try {
     ov::Tensor image = pipe.generate(prompt,
         ov::genai::width(512),
         ov::genai::height(512),
-        ov::genai::num_inference_steps(20),
-        ov::genai::num_images_per_prompt(1));
+        ov::genai::num_inference_steps(20));
```
Comment (likholat) [inline samples/cpp/text2image/main.cpp:18]: Shouldn't we show the usage of `num_images_per_prompt` param there?

**Review by likholat — APPROVED** [review]

File: src/docs/SUPPORTED_MODELS.md [hunk 2]
```diff
@@ -178,6 +178,45 @@ The pipeline can work with other similar topologies produced by `optimum-intel`
...
+    <tr>
+      <td><code>Latent Consistency Model</code></td>
+      <td>
+        <ul>
+          <li><a href="https://huggingface.co/SimianLuo/LCM_Dreamshaper_v7"><code>SimianLuo/LCM_Dreamshaper_v7</code></a></li>
+        </ul>
+      </td>
+    </tr>
+    <tr>
+      <td><code>Stable Diffusion XL</code></td>
```
Comment (Wovchena) [inline src/docs/SUPPORTED_MODELS.md:209]: Sorting the table alphabetically would be nice :) Maybe it's worth putting `Text 2 image models` before `Visual language models` for the same reason.

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: .github/workflows/causal_lm_cpp.yml [hunk 3]
```diff
@@ -258,7 +258,7 @@ jobs:
           && python samples\python\greedy_causal_lm\greedy_causal_lm.py .\TinyLlama-1.1B-Chat-v1.0\ 69 > .\py.txt
       - run: fc .\cpp.txt .\py.txt
 
-  cpp-beam_search_causal_lm-Qwen-7B-Chat:
+  cpp-greedy_causal_lm-Qwen-7B-Chat:
```
[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply .github/workflows/causal_lm_cpp.yml:257]: good catch, fixed

File: samples/cpp/text2image/main.cpp (hunk 1 above)
Comment (ilya-lavrenov) [reply samples/cpp/text2image/main.cpp]: reverted

File: src/docs/SUPPORTED_MODELS.md (hunk 2 above)
Comment (ilya-lavrenov) [reply src/docs/SUPPORTED_MODELS.md]: done

**Review by Wovchena — APPROVED** [review]

File: samples/cpp/text2image/imwrite.cpp [hunk 4]
```diff
@@ -131,3 +130,19 @@ void imwrite(const std::string& name, ov::Tensor image, bool convert_bgr2rgb) {
         output_file.write(reinterpret_cast<const char*>(pad), padSize);
     }
 }
+
+} // namespace
+
+
+void imwrite(const std::string& name, ov::Tensor images, bool convert_bgr2rgb) {
+    const ov::Shape shape = images.get_shape(); img_shape = {1, shape[1], shape[2], shape[3]};
```
Comment (ilya-lavrenov) [inline samples/cpp/text2image/imwrite.cpp:138]:
```suggestion
    const ov::Shape shape = images.get_shape(), img_shape = {1, shape[1], shape[2], shape[3]};
```

**Pull request merged**